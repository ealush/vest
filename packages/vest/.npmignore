src
playground
config
testUtils
./jest.config.js
dist