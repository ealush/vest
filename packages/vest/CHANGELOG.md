# vest - Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## 1.0.31 - 2021-07-02
### Fixed and improved
- e6ea7d6  support global replaces (undefined)
- 34e0414  improved conditions (undefined)
- 26c28c6  all tests pass (undefined)
- 33f4e46  release (undefined)
- 6fe40c7  better bundle (undefined)
- c2cfb65  better typing (undefined)
- c6387ab  before ts settings (undefined)
- c0e9708  generate correct d.ts file (undefined)
- packages/vest/types/vest.development.d.ts
- 8e01b8e  x (undefined)
- afb3960  x (undefined)
- e0a8463  add changelog support (undefined)
- cc46c38  current (undefined)
- 4a2073f  patch(vest): changes (ealush)
- c0a1705  patch(vest): changes (ealush)
- 240386c  better create generic (ealush)
- b6db1c6  transform any to unknowns (ealush)
- f2c20a7  add done integration tests (ealush)
- ba56e33  produce methods (ealush)
- 0638243  better produce integration tests (ealush)
- 4098a11  getFailures as integration tests (ealush)
- f56cf7f  pd (ealush)
- 2ea440e  pd (ealush)
- b3032ef  relocate draftresult (ealush)
- 962141f  test getFailuresByGroup (ealush)
- e163c56  test getFailures (ealush)
- a92b6df  test collectFailureMessages (ealush)
- 81aad51  fix most tests (ealush)
- c149e90  simplify collectFailureMessages (ealush)
