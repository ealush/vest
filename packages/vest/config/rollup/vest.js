const addEsmDir = require('../../../../config/rollup/addEsmDir');
const genConfig = require('../../../../config/rollup/genConfig');
const writeMainTemplate = require('../../../../config/rollup/writeMainTemplate');
const { packageDist, packageNames } = require('../../../../util');

addEsmDir(packageDist(packageNames.VEST));
writeMainTemplate(packageDist(packageNames.VEST), packageNames.VEST);

export default genConfig({
  distPath: packageDist(packageNames.VEST),
  esInput: 'vest.mjs.js',
  input: 'vest.js',
  libraryName: packageNames.VEST,
});
