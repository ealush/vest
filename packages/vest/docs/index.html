<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>Vest - Validation Test</title>
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="description" content="Validation Test" />
    <meta
      name="viewport"
      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
    />
    <link rel="icon" href="./_assets/favicon.ico" />
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/npm/docsify-themeable@0/dist/css/theme-simple.css"
    />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/prismjs@1.22.0/themes/prism-tomorrow.css">
  </head>
  <body>
    <div id="app"></div>
    <script>
      window.$docsify = {
        name: 'vest',
        logo: '/_assets/vest-logo.png',
        repo: 'https://github.com/ealush/vest',
        search: 'auto',
        loadSidebar: true,
        subMaxLevel: 2,
        themeColor: '#5BA9D8',
      };
    </script>
    <script src="//unpkg.com/docsify/lib/docsify.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/docsify-themeable@0"></script>
    <script src="//unpkg.com/docsify/lib/plugins/search.min.js"></script>
    <script src="//unpkg.com/vest"></script>
    <script src="./_assets/index.js"></script>
  </body>
</html>
