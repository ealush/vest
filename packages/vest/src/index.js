import any from "anyone/any";
import enforce from "n4s/dist/enforce.min";
import { VERSION } from "./constants";
import test from "./core/test";
import validate from "./core/validate";
import { draft, only, skip, warn } from "./hooks";
import { singleton, runWithContext } from "./lib";

export default singleton.register({
  Enforce: enforce.Enforce,
  VERSION,
  any,
  draft,
  enforce,
  only,
  runWithContext,
  skip,
  test,
  validate,
  warn,
});
