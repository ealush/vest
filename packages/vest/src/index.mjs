import enforce from '../../n4s/src/enforce';
import create from './core/suite/create';
import test from './core/test';
import { draft, only, skip, warn, group } from './hooks';

const VERSION = VEST_VERSION;

export default {
  VERSION,
  create,
  draft,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};

export { VERSION, create, draft, enforce, group, only, skip, test, warn };
