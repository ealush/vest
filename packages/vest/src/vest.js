import create from 'createSuite';
import enforce from 'enforce';
import { only, skip, warn, group } from 'hooks';
import test from 'test';

const VERSION = __LIB_VERSION__;

export default {
  VERSION,
  create,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};
