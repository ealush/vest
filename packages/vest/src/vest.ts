import { enforce } from 'n4s';

import create from 'create';
import context from 'ctx';
import each from 'each';
import { only, skip } from 'exclusive';
import group from 'group';
import include from 'include';
import { eager } from 'mode';
import omitWhen from 'omitWhen';
import optional from 'optionalTests';
import skipWhen from 'skipWhen';
import { test } from 'test';
import warn from 'warn';

const VERSION = __LIB_VERSION__;
export {
  test,
  create,
  each,
  only,
  skip,
  warn,
  group,
  optional,
  skipWhen,
  omitWhen,
  enforce,
  VERSION,
  context,
  include,
  eager,
};
