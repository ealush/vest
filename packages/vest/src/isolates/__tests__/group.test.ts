import * as vest from 'vest';

enum GroupNames {
  G1 = 'g1',
  G2 = 'g2',
}

enum FieldNames {
  F1 = 'f1',
  F2 = 'f2',
  F3 = 'f3',
  F4 = 'f4',
}

describe('group', () => {
  it('should run group callback', () => {
    const groupName = 'groupName';
    const callback = jest.fn();
    vest.create(() => {
      vest.group(groupName, callback);
    })();
    expect(callback).toHaveBeenCalled();
  });

  it('Should run the tests within the group', () => {
    const cb1 = jest.fn(() => false);
    const cb2 = jest.fn(() => false);
    const cb3 = jest.fn(() => {});
    const suite = vest.create(() => {
      vest.mode(vest.Modes.ALL);
      vest.test(FieldNames.F1, () => false);

      vest.group(GroupNames.G1, () => {
        vest.test(FieldNames.F1, cb1);
        vest.test(FieldNames.F2, cb2);
        vest.test(FieldNames.F3, cb3);
      });
    });

    const res = suite();
    expect(res.hasErrors(FieldNames.F1)).toBe(true);
    expect(res.hasErrors(FieldNames.F2)).toBe(true);
    expect(res.hasErrors(FieldNames.F3)).toBe(false);
    expect(res.hasErrorsByGroup(GroupNames.G1)).toBe(true);
    expect(res.hasErrorsByGroup(GroupNames.G1, FieldNames.F1)).toBe(true);
    expect(res.hasErrorsByGroup(GroupNames.G1, FieldNames.F2)).toBe(true);
    expect(res.hasErrorsByGroup(GroupNames.G1, FieldNames.F3)).toBe(false);
    expect(cb1).toHaveBeenCalled();
    expect(cb2).toHaveBeenCalled();
    expect(cb3).toHaveBeenCalled();
    expect(res.isValid()).toBe(false);
    expect(res.isValidByGroup(GroupNames.G1)).toBe(false);
    expect(res.isValidByGroup(GroupNames.G1, FieldNames.F1)).toBe(false);
    expect(res.isValidByGroup(GroupNames.G1, FieldNames.F2)).toBe(false);
    expect(res.isValidByGroup(GroupNames.G1, FieldNames.F3)).toBe(true);
    expect(res.tests[FieldNames.F1].testCount).toBe(2);
    expect(res.tests[FieldNames.F2].testCount).toBe(1);
    expect(res.tests[FieldNames.F3].testCount).toBe(1);
    expect(res.tests[FieldNames.F1].errorCount).toBe(2);
    expect(res.tests[FieldNames.F2].errorCount).toBe(1);
    expect(res.tests[FieldNames.F3].errorCount).toBe(0);
    expect(res.groups[GroupNames.G1][FieldNames.F1].errorCount).toBe(1);
    expect(res.groups[GroupNames.G1][FieldNames.F1].testCount).toBe(1);
    expect(res.groups[GroupNames.G1][FieldNames.F2].errorCount).toBe(1);
    expect(res.groups[GroupNames.G1][FieldNames.F2].testCount).toBe(1);
    expect(res.groups[GroupNames.G1][FieldNames.F3].errorCount).toBe(0);
    expect(res.groups[GroupNames.G1][FieldNames.F3].testCount).toBe(1);
    expect(suite.get()).toMatchSnapshot();
  });

  describe('Multiple groups', () => {
    it('Should run the tests within the groups', () => {
      const cb1 = jest.fn(() => false);
      const cb2 = jest.fn(() => false);
      const cb3 = jest.fn(() => {});
      const suite = vest.create(() => {
        vest.mode(vest.Modes.ALL);
        vest.test(FieldNames.F1, () => false);

        vest.group(GroupNames.G1, () => {
          vest.test(FieldNames.F1, cb1);
          vest.test(FieldNames.F2, cb2);
          vest.test(FieldNames.F3, cb3);
        });

        vest.group(GroupNames.G2, () => {
          vest.test(FieldNames.F1, cb1);
          vest.test(FieldNames.F2, cb2);
          vest.test(FieldNames.F3, cb3);
        });
      });

      const res = suite();
      expect(res.hasErrors(FieldNames.F1)).toBe(true);
      expect(res.hasErrors(FieldNames.F2)).toBe(true);
      expect(res.hasErrors(FieldNames.F3)).toBe(false);
      expect(res.hasErrorsByGroup(GroupNames.G1)).toBe(true);
      expect(res.hasErrorsByGroup(GroupNames.G1, FieldNames.F1)).toBe(true);
      expect(res.hasErrorsByGroup(GroupNames.G1, FieldNames.F2)).toBe(true);
      expect(res.hasErrorsByGroup(GroupNames.G1, FieldNames.F3)).toBe(false);
      expect(res.hasErrorsByGroup(GroupNames.G2)).toBe(true);
      expect(res.hasErrorsByGroup(GroupNames.G2, FieldNames.F1)).toBe(true);
      expect(res.hasErrorsByGroup(GroupNames.G2, FieldNames.F2)).toBe(true);
      expect(res.hasErrorsByGroup(GroupNames.G2, FieldNames.F3)).toBe(false);
      expect(cb1).toHaveBeenCalledTimes(2);
      expect(cb2).toHaveBeenCalledTimes(2);
      expect(cb3).toHaveBeenCalledTimes(2);
      expect(res.isValid()).toBe(false);
      expect(res.isValidByGroup(GroupNames.G1)).toBe(false);
      expect(res.isValidByGroup(GroupNames.G1, FieldNames.F1)).toBe(false);
      expect(res.isValidByGroup(GroupNames.G1, FieldNames.F2)).toBe(false);
      expect(res.isValidByGroup(GroupNames.G1, FieldNames.F3)).toBe(true);
      expect(res.isValidByGroup(GroupNames.G2)).toBe(false);

      expect(res.isValidByGroup(GroupNames.G2, FieldNames.F1)).toBe(false);
      expect(res.isValidByGroup(GroupNames.G2, FieldNames.F2)).toBe(false);
      expect(res.isValidByGroup(GroupNames.G2, FieldNames.F3)).toBe(true);
      expect(res.tests[FieldNames.F1].testCount).toBe(3);
      expect(res.tests[FieldNames.F2].testCount).toBe(2);
      expect(res.tests[FieldNames.F3].testCount).toBe(2);
      expect(res.tests[FieldNames.F1].errorCount).toBe(3);
      expect(res.tests[FieldNames.F2].errorCount).toBe(2);
      expect(res.tests[FieldNames.F3].errorCount).toBe(0);
      expect(res.groups[GroupNames.G1][FieldNames.F1].errorCount).toBe(1);
      expect(res.groups[GroupNames.G1][FieldNames.F1].testCount).toBe(1);
      expect(res.groups[GroupNames.G1][FieldNames.F2].errorCount).toBe(1);
      expect(res.groups[GroupNames.G1][FieldNames.F2].testCount).toBe(1);
      expect(res.groups[GroupNames.G1][FieldNames.F3].errorCount).toBe(0);
      expect(res.groups[GroupNames.G1][FieldNames.F3].testCount).toBe(1);
      expect(res.groups[GroupNames.G2][FieldNames.F1].errorCount).toBe(1);
      expect(res.groups[GroupNames.G2][FieldNames.F1].testCount).toBe(1);
      expect(res.groups[GroupNames.G2][FieldNames.F2].errorCount).toBe(1);
      expect(res.groups[GroupNames.G2][FieldNames.F2].testCount).toBe(1);
      expect(res.groups[GroupNames.G2][FieldNames.F3].errorCount).toBe(0);
      expect(res.groups[GroupNames.G2][FieldNames.F3].testCount).toBe(1);
      expect(suite.get()).toMatchSnapshot();
    });
  });
});
