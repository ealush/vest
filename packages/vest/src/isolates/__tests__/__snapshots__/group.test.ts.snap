// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group Multiple groups Should run the tests within the groups 1`] = `
SuiteSummary {
  "errorCount": 5,
  "errors": [
    SummaryFailure {
      "fieldName": "f1",
      "groupName": undefined,
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "f1",
      "groupName": "g1",
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "f2",
      "groupName": "g1",
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "f1",
      "groupName": "g2",
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "f2",
      "groupName": "g2",
      "message": undefined,
    },
  ],
  "getError": [Function],
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarning": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": {
    "g1": {
      "f1": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "f2": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "f3": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "testCount": 1,
        "valid": true,
        "warnCount": 0,
        "warnings": [],
      },
    },
    "g2": {
      "f1": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "f2": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "f3": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "testCount": 1,
        "valid": true,
        "warnCount": 0,
        "warnings": [],
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "isValid": [Function],
  "isValidByGroup": [Function],
  "suiteName": undefined,
  "testCount": 7,
  "tests": {
    "f1": {
      "errorCount": 3,
      "errors": [],
      "testCount": 3,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
    "f2": {
      "errorCount": 2,
      "errors": [],
      "testCount": 2,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
    "f3": {
      "errorCount": 0,
      "errors": [],
      "testCount": 2,
      "valid": true,
      "warnCount": 0,
      "warnings": [],
    },
  },
  "valid": false,
  "warnCount": 0,
  "warnings": [],
}
`;

exports[`group Should run the tests within the group 1`] = `
SuiteSummary {
  "errorCount": 3,
  "errors": [
    SummaryFailure {
      "fieldName": "f1",
      "groupName": undefined,
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "f1",
      "groupName": "g1",
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "f2",
      "groupName": "g1",
      "message": undefined,
    },
  ],
  "getError": [Function],
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarning": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": {
    "g1": {
      "f1": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "f2": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "f3": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "testCount": 1,
        "valid": true,
        "warnCount": 0,
        "warnings": [],
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "isValid": [Function],
  "isValidByGroup": [Function],
  "suiteName": undefined,
  "testCount": 4,
  "tests": {
    "f1": {
      "errorCount": 2,
      "errors": [],
      "testCount": 2,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
    "f2": SummaryBase {
      "errorCount": 1,
      "errors": [],
      "testCount": 1,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
    "f3": SummaryBase {
      "errorCount": 0,
      "errors": [],
      "testCount": 1,
      "valid": true,
      "warnCount": 0,
      "warnings": [],
    },
  },
  "valid": false,
  "warnCount": 0,
  "warnings": [],
}
`;

exports[`group: base case Group validity 1`] = `
{
  "done": [Function],
  "errorCount": 4,
  "errors": [
    SummaryFailure {
      "fieldName": "field_2",
      "groupName": "group_1",
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "field_1",
      "groupName": "group_2",
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "field_1",
      "groupName": "group_3",
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "field_2",
      "groupName": "group_3",
      "message": undefined,
    },
  ],
  "getError": [Function],
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarning": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": {
    "group_1": {
      "field_1": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "testCount": 1,
        "valid": true,
        "warnCount": 0,
        "warnings": [],
      },
      "field_2": {
        "errorCount": 1,
        "errors": [],
        "testCount": 2,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
    },
    "group_2": {
      "field_1": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "field_2": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "testCount": 1,
        "valid": true,
        "warnCount": 0,
        "warnings": [],
      },
    },
    "group_3": {
      "field_1": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "field_2": SummaryBase {
        "errorCount": 1,
        "errors": [],
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
    },
    "group_4": {
      "field_1": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "testCount": 1,
        "valid": true,
        "warnCount": 0,
        "warnings": [],
      },
      "field_2": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "testCount": 1,
        "valid": true,
        "warnCount": 0,
        "warnings": [],
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "isValid": [Function],
  "isValidByGroup": [Function],
  "suiteName": undefined,
  "testCount": 9,
  "tests": {
    "field_1": {
      "errorCount": 2,
      "errors": [],
      "testCount": 4,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
    "field_2": {
      "errorCount": 2,
      "errors": [],
      "testCount": 5,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
  },
  "valid": false,
  "warnCount": 0,
  "warnings": [],
}
`;
